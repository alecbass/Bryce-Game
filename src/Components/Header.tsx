import * as React from "react";
import styled from "styled-components";

const Container = styled("div")`
    display: flex;
    height: 16%;
    width: 100%;
    align-items: center;
    justify-content: center;
`;

interface Props {
    title: string;
    style?: React.CSSProperties;
}

class Header extends React.PureComponent<Props> {
    static defaultProps: Partial<Props> = {
        style: {
            backgroundColor: "lightblue"
        }
    };

    render() {
        const { title, style } = this.props;

        return (
            <Container style={style}>
                <span>{title}</span>
            </Container>
        );
    }
}

export default Header;