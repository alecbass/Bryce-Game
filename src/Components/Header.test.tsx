import * as React from "react";
import { render, fireEvent, cleanup, act, wait } from "react-testing-library";
import "jest-dom/extend-expect";

import Header from "./Header";

const defaultProps: {
    title: string;
    style?: React.CSSProperties;
} = {
    title: "Main Header",
    style: {
        backgroundColor: "red"
    }
};

afterEach(cleanup);

it("creates a header with text", () => {
    const { container, getByText } = render(
        <Header title={defaultProps.title} style={defaultProps.style} />
    );
    // title
    const testTitle = getByText(defaultProps.title);
    expect(testTitle).toBeInTheDocument();
    expect(testTitle).toBeInstanceOf(HTMLSpanElement);
    expect((testTitle as HTMLSpanElement).innerHTML).toBe(defaultProps.title);

    // surrounding div container
    expect(container).toBeInstanceOf(HTMLDivElement);
    const div = container.firstChild as HTMLDivElement;
    expect(div).toBeInstanceOf(HTMLDivElement);
    expect(div).toBeEmpty;
    expect(div).toHaveStyle(`background-color: ${defaultProps.style!.backgroundColor}`);
    expect(div).toHaveStyle("height: 16%");
    expect(container).toMatchSnapshot(`
        <div>
            <span>Main Header</span>
        </div>
    `);
});