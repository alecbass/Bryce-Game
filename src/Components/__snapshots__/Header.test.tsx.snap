// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a header with text: 
        <div>
            <span>Main Header</span>
        </div>
     1`] = `
<div>
  <div
    class="sc-bdVaJa eQDQwl"
    style="background-color: red;"
  >
    <span>
      Main Header
    </span>
  </div>
</div>
`;
